Updates

abdul-sayed22 opened the pull request:
The following PR contains updates as follows:

## 1. Drawable background track
You are now able to set the background track of the graph to a drawable.

`app:backgroundTrackDrawable="@drawable/my_background"`

The drawable will take precedence over the solid colour.

![image](https://user-images.githubusercontent.com/52398157/94911829-01223280-04a7-11eb-9fab-a036666729bd.png)


## 2. Programmable toggling
You are now able to programmatically toggle the following components of the graph: Graph Node, Background Track

`graph.setBackgroundTrack(newColor)`
`graph.setBackgroundTrack(newDrawable)`
`graph.removeBackgroundTrack()`

`graph.setGraphNode(newNodeType)`

## 3. Animation
Due to the above changes (2), you will sometimes need to redraw the graph. You can now use the following functions to achieve that.

`graph.redraw(data)`
`graph.animateIn()` (already present)
`graph.animateOut()` (new)

## 4. Example App Enhancements
New buttons have been added to the example app interface that allows you to toggle certain settings and play around with the graph.

![image](https://user-images.githubusercontent.com/52398157/94911266-295d6180-04a6-11eb-95ee-80ce28d257da.png)

abdul-sayed22 committed:
Merge branch 'master' into master-fork

abdul-sayed22 committed:
Auto stash before merge of "master" and "origin/master"

abdul-sayed22 committed:
Minor fixes after merging remote branch

abdul-sayed22 committed:
Clean up

abdul-sayed22 committed:
Add animate out function

abdul-sayed22 committed:
Allow to toggle graph node and background track programmatically

abdul-sayed22 committed:
Enhance example app

abdul-sayed22 committed:
Add button to play with animations

abdul-sayed22 committed:
Toggle safety

DuartBreedt commented on radialgraph/src/main/java/com/duartbreedt/radialgraph/view/RadialGraph.kt:
```diff
@@ -136,9 +139,40 @@ class RadialGraph : ConstraintLayout {
+        graphConfig.backgroundTrackDrawable = newDrawable
+    }
+
+    fun removeBackgroundTrack() {
+        with (graphConfig) {
+            backgroundTrackColor = NO_ID
+            backgroundTrackDrawable = null
+        }
+    }
```
Won't `setBackgroundTrack(null)` automatically do this? I feel this function may be unncessary.

abdul-sayed22 replied:
The problem here is that `setBackgroundTrack` has 2 signatures and cannot resolve when you pass `null`

DuartBreedt replied:
If you qualify either of the parameters:
`setBackgroundTrack(newDrawable = null)` or `setBackgroundTrack(newColor = null)`
It will work, so it's still workable, or do you disagree?

abdul-sayed22 replied:
While workable, it might not be intuitive that this is a possibility. That could be remedied by adding kdocs to the `setBackgroundTrack` function though, so I will settle on that route

DuartBreedt commented on radialgraph/src/main/java/com/duartbreedt/radialgraph/view/RadialGraph.kt:
```diff
@@ -136,9 +139,40 @@ class RadialGraph : ConstraintLayout {
+        }
+    }
+
+    fun redraw(data: Data, animateIn: Boolean = true) {
+        setData(data)
+        if (animateIn) animateIn()
+    }
```
I am not sure if this is needed. I'd argue if someone needs to redraw it they should call these functions themselves. 
For instance, what if you want to animateOut first then animateIn the new data? I feel users of this library can extend the graph and add these compound functions in themselves as they are needed.

abdul-sayed22 replied:
Makes sense. I will remove this from the library then

abdul-sayed22 committed:
Address PR comments

abdul-sayed22 committed:
Remove the removeBackgroundTrack function in lieu of existing functions

The addition of kdocs to that function allows an end user to see the additional capability of setBackgroundTrack functions
